packages/node_modules/glimmer-runtime/lib/compiled/opcodes/dom.ts

```diff
@@ -87,27 +91,29 @@ export class CloseElementOpcode extends Opcode {
 
     let classes = new ClassList();
     let flattened = dict<ElementOperation>();
+    let flattenedKeys = [];
 
-    groups.forEach(group => {
-      group.forEach(op => {
+    for (let i = 0; i < groups.length; i++) {
+      for (let j = 0; j < groups[i].length; j++) {
+        let op = groups[i][j];
         let name: string = <FIXME> op['name'];
         let value: PathReference = <FIXME> op['value'];
-
         if (name === 'class') {
           classes.append(value);
         } else {
+          flattenedKeys.push(name);
           flattened[name] = op;
         }
-      });
-    });
+      }
+    }
 
     if (!classes.isEmpty()) {
       vm.updateWith(new NonNamespacedAttribute('class' as InternedString, classes).flush(dom, element));
     }
 
-    Object.keys(flattened).forEach(key => {
-      vm.updateWith(flattened[key].flush(dom, element));
-    });
+    for (let i = 0; i < flattenedKeys.length; i++) {
```

Comment:
if you let i the same name the de-conflicted name will not be correct in the source map names, so this hurts debugging, also, it can cause crazy code output from babel.